src/qml/qmlapplication.cpp

```diff
@@ -58,6 +68,25 @@ QmlApplication::QmlApplication(
     // singletons to that they can be accessed by components instantiated by
     // QML, which would also be suboptimal.
 
+    qmlRegisterSingletonType<QmlDlgPreferencesProxy>("Mixxx",
+            0,
+            1,
+            "PreferencesDialog",
+            lambda_to_singleton_type_factory_ptr(
+                    [this](QQmlEngine* pEngine,
```

Comment:
I also tried making the DlgPreferences instance a local variable of the QmlApplication constructor and a local variable within this lambda to test if capturing `this` was causing the problem, but neither of those solved the problem.